app/controllers/sign_up/completions_controller.rb

```diff
@@ -0,0 +1,24 @@
+module SignUp
+  class CompletionsController < ApplicationController
```

Comment:
Or maybe it's OK to always let them see this page, as long as `session[:sp]` is present?